src/uct/ib/base/ib_md.c

```diff
@@ -580,8 +580,8 @@ ucs_status_t uct_ib_memh_alloc(uct_ib_md_t *md, size_t length,
     memh->lkey = UCT_IB_INVALID_MKEY;
     memh->rkey = UCT_IB_INVALID_MKEY;
 
-    if ((mem_flags & UCT_MD_MEM_FLAG_NONBLOCK) && (length > 0) &&
-        (md->reg_nonblock_mem_types & UCS_BIT(UCS_MEMORY_TYPE_HOST))) {
+    if ((mem_flags & UCT_MD_MEM_FLAG_NONBLOCK) && (length > 0)) {
+        /* Registration will fail if memory does not actually support it */
```

Comment:
need to re-add the ODP check as we see related failures. I think it makes sense to check if odp as a whole is enabled, before adding  it.

Reply:
does it mean UCT users (UCP or gtest) pass NONBLOCK flag even though it is not supported?

Reply:
yes

Reply:
so shouldn't we fix the upper layers?

Reply:
i am assuming this would be handled by https://github.com/openucx/ucx/pull/10079

Reply:
ok